timeservice.go

```diff
@@ -184,98 +227,130 @@ func newDaemonConnector(ctx context.Context, log *zap.Logger, daemonAddr string)
 				udp.UDPAddrFromSnet(localAddr),
 				udp.UDPAddrFromSnet(remoteAddr),
+				cfg.AuthMode,
+				s,
+				cfg.NTSKEInsecureSkipVerify,
 			))
 			dstIAs = append(dstIAs, remoteAddr.IA)
+		} else {
+			refClocks = append(refClocks, newNTPRefernceClockIP(
+				localAddr.Host,
+				remoteAddr.Host,
+				cfg.AuthMode,
+				s,
+				cfg.NTSKEInsecureSkipVerify,
+			))
 		}
-		if daemonAddr != "" {
-			ctx := context.Background()
-			pather := scion.StartPather(ctx, log, daemonAddr, dstIAs)
-			drkeyFetcher := scion.NewFetcher(newDaemonConnector(ctx, log, daemonAddr))
-			for _, c := range refClocks {
-				scionclk, ok := c.(*ntpReferenceClockSCION)
-				if ok {
-					scionclk.pather = pather
-					for i := 0; i != len(scionclk.ntpcs); i++ {
-						scionclk.ntpcs[i].Auth.Enabled = true
-						scionclk.ntpcs[i].Auth.DRKeyFetcher = drkeyFetcher
-					}
+	}
+
+	for _, s := range cfg.SCIONPeers {
+		remoteAddr, err := snet.ParseUDPAddr(s)
+		if err != nil {
+			log.Fatal("failed to parse peer address", zap.String("address", s), zap.Error(err))
+		}
+		if remoteAddr.IA.IsZero() {
+			log.Fatal("unexpected peer address", zap.String("address", s), zap.Error(err))
+		}
+		netClocks = append(netClocks, newNTPRefernceClockSCION(
+			udp.UDPAddrFromSnet(localAddr),
+			udp.UDPAddrFromSnet(remoteAddr),
+			cfg.AuthMode,
+			s,
+			cfg.NTSKEInsecureSkipVerify,
+		))
+		dstIAs = append(dstIAs, remoteAddr.IA)
+	}
+
+	if cfg.DaemonAddress != "" {
+		ctx := context.Background()
+		pather := scion.StartPather(ctx, log, cfg.DaemonAddress, dstIAs)
+		drkeyFetcher := scion.NewFetcher(newDaemonConnector(ctx, log, cfg.DaemonAddress))
+		for _, c := range refClocks {
+			scionclk, ok := c.(*ntpReferenceClockSCION)
+			if ok {
+				scionclk.pather = pather
+				for i := 0; i != len(scionclk.ntpcs); i++ {
+					scionclk.ntpcs[i].Auth.Enabled = true
+					scionclk.ntpcs[i].Auth.DRKeyFetcher = drkeyFetcher
 				}
 			}
-			for _, c := range netClocks {
-				scionclk, ok := c.(*ntpReferenceClockSCION)
-				if ok {
-					scionclk.pather = pather
-					for i := 0; i != len(scionclk.ntpcs); i++ {
-						scionclk.ntpcs[i].Auth.Enabled = true
-						scionclk.ntpcs[i].Auth.DRKeyFetcher = drkeyFetcher
-					}
+		}
+		for _, c := range netClocks {
+			scionclk, ok := c.(*ntpReferenceClockSCION)
+			if ok {
+				scionclk.pather = pather
+				for i := 0; i != len(scionclk.ntpcs); i++ {
+					scionclk.ntpcs[i].Auth.Enabled = true
+					scionclk.ntpcs[i].Auth.DRKeyFetcher = drkeyFetcher
 				}
 			}
 		}
 	}
+
 	return
 }
 
-func loadNTSKEConfig(log *zap.Logger, configFile string) *tls.Config {
-	var cfg svcConfig
-	raw, err := os.ReadFile(configFile)
-	if err != nil {
-		log.Fatal("failed to load configuration", zap.Error(err))
-	}
-	err = toml.NewDecoder(bytes.NewReader(raw)).DisallowUnknownFields().Decode(&cfg)
-	if err != nil {
-		log.Fatal("failed to decode configuration", zap.Error(err))
-	}
-
+func tlsConfig(log *zap.Logger, cfg svcConfig) *tls.Config {
```

Comment:
I would probably remove the empty lines in this function.

Reply:
Done